File: src/main/java/com/mysettlement/domain/video/entity/Video.java
package com.mysettlement.domain.video.entity;

import com.mysettlement.domain.user.entity.User;
import com.mysettlement.domain.video.dto.request.VideoStatusChangeRequest;
import com.mysettlement.domain.video.dto.request.VideoUpdateRequest;
import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;

import static jakarta.persistence.EnumType.STRING;
import static jakarta.persistence.FetchType.LAZY;
import static lombok.AccessLevel.PROTECTED;

@Entity
@Getter
@Table(name = "VIDEO")
@NoArgsConstructor(access = PROTECTED)
public class Video {

    @Id
    @Column(name = "video_id")
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne(fetch = LAZY)
    @JoinColumn(name = "user_id")
    private User user;

    @Column(name = "video_title")
    private String videoTitle;

    @Column(name = "video_desc")
    private String videoDesc;

    @Column(name = "video_path")
    private String videoPath;

    @Column(name = "video_view")
    private long videoView = 0;

    @Column(name = "video_price_per_view")
    private double videoPricePerView = 1.0;

    @Column(name = "video_status")
    @Enumerated(STRING)
    private VideoStatus videoStatus;

    @Column(name="video_size")
    private long videoSize;

    @Builder
    private Video(User user, String videoTitle, String videoDesc, String videoPath, VideoStatus videoStatus, long videoSize) {
        this.user = user;
        this.videoTitle = videoTitle;
        this.videoDesc = videoDesc;
        this.videoPath = videoPath;
        this.videoStatus = videoStatus;
        this.videoSize = videoSize;
    }

    public void updateStatus(VideoStatusChangeRequest videoStatusChangeRequest) {
        this.videoStatus = videoStatusChangeRequest.videoStatus();
    }

    public void updateStatus(VideoUpdateRequest videoUpdateRequest) {
        this.videoTitle = videoUpdateRequest.title();
        this.videoDesc = videoUpdateRequest.desc();
    }

    public void viewUpdate() {
        this.videoView++;
    }

    public void changeInfoWith(VideoUpdateRequest videoUpdateRequest) {
        this.videoTitle = videoUpdateRequest.title();
        this.videoDesc = videoUpdateRequest.desc();
    }
}

File: front/src/entities/video.ts
import type { UserResponseData } from '@/entities/user.ts'
import User from '@/entities/user.ts'

export interface VideoResponseData {
  id: number
  title: string
  user: UserResponseData
  desc: string
  videoStatus: string
  videoView: number
}

export default class Video {
  private readonly _id: number
  private readonly _title: string
  private readonly _user: User
  private readonly _desc: string
  private readonly _videoStatus: string
  private readonly _videoView: number

  constructor(
    id: number,
    title: string,
    user: User,
    desc: string,
    videoStatus: string,
    videoView: number,
  ) {
    this._id = id
    this._title = title
    this._user = user
    this._desc = desc
    this._videoStatus = videoStatus
    this._videoView = videoView
  }

  public static of(data: VideoResponseData): Video {
    return new Video(
      data.id,
      data.title,
      User.of(data.user),
      data.desc,
      data.videoStatus,
      data.videoView,
    )
  }

  get id(): number {
    return this._id
  }

  get title(): string {
    return this._title
  }

  get user(): User {
    return this._user
  }

  get desc(): string {
    return this._desc
  }

  get status(): string {
    return this._videoStatus
  }

  get videoView(): number {
    return this._videoView
  }
}
